### PR TITLE
feat: #107: 24h stats design

### LINKED ISSUE #107: Add design to 24hr stats
Screenshot 2024-10-24 at 4 34 59 PM

Reference: https://www.figma.com/design/xWsVprkEgEihweVN9isrGQ/DEX-(Incl.-UI-Lib)?node-id=3658-145100&node-type=canvas&t=NYogohDveZXB9V4J-0

### LINKED ISSUE #108: Loading skeleton of 24hr stats
Screenshot 2024-10-24 at 4 36 32 PM

Reference: https://www.figma.com/design/xWsVprkEgEihweVN9isrGQ/DEX-(Incl.-UI-Lib)?node-id=3658-145100&node-type=canvas&t=NYogohDveZXB9V4J-0

### PR DESCRIPTION
**Pull request by VanishMax**
Closes #107 

Closes #108 

Added styles and some neat animations for the stats block. Also, did some preliminary work towards #131 

image

And mobile:

image

Animations:

https://github.com/user-attachments/assets/d1a775b1-d97f-4d16-b357-cd5dbc3058b9

### CODE REVIEW

File: src/pages/trade/ui/summary.tsx [hunk 1]
```diff
@@ -1,24 +1,120 @@
...
   const { data, isLoading, error } = useSummary();
+  const { quoteAsset } = usePathToMetadata();
 
-  if (error) {
-    return <div className='text-red-600'>Error: {String(error)}</div>;
-  }
+  const change24h = data && {
+    positive: data.current_price >= data.price_24h_ago,
+    change: Number(data.current_price - data.price_24h_ago).toFixed(4),
+    percent: Number(
```
Comment (JasonMHasperhoven) [inline src/pages/trade/ui/summary.tsx:52]: Same here.

File: src/pages/trade/ui/summary.tsx [hunk 2]
```diff
@@ -1,24 +1,120 @@
...
-      <div>low_24h: {data.low_24h}</div>
-      <div>high_24h: {data.high_24h}</div>
-      <div>price_24h_ago: {data.price_24h_ago}</div>
-      <div>swap_volume_24h: {data.swap_volume_24h}</div>
-      <div>direct_volume_24h: {data.direct_volume_24h}</div>
+    <div className='flex flex-wrap items-center gap-x-4 gap-y-2'>
+      <SummaryCard title='Price' loading={isLoading}>
+        {data && (
+          <Text detail color='text.primary'>
+            {Number(data.current_price).toFixed(6)}
```
Comment (JasonMHasperhoven) [inline src/pages/trade/ui/summary.tsx:72]: And here.

File: src/pages/trade/ui/summary.tsx [hunk 3]
```diff
@@ -1,24 +1,120 @@
...
+                {change24h.percent}%
+              </Text>
+            </span>
+          </div>
+        )}
+      </SummaryCard>
+      <SummaryCard title='24h High' loading={isLoading}>
+        {data && (
+          <Text detail color='text.primary'>
+            {Number(data.high_24h).toFixed(4)}
```
Comment (JasonMHasperhoven) [inline src/pages/trade/ui/summary.tsx:97]: And here and so forth, we have the round fn in shared/utils

File: src/shared/utils/numbers/shortify.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Makes large numbers shorter. Examples:
+ * - 999 -> 999
+ * - 1000 -> 1K
+ * - 1000000 -> 1M
+ * - 1000000000 -> 1B
+ */
+export const shortify = (value: number): string => {
```
Comment (JasonMHasperhoven) [inline src/shared/utils/numbers/shortify.ts:8]: There's a package `friendly-numbers`. Perhaps it's more wise to use that.

Comment (VanishMax) [reply src/shared/utils/numbers/shortify.ts:8]: But it does just the same job, only the 'T' case (trillions) is not covered by me. Why would we need a library for that?

File: src/pages/trade/ui/summary.tsx [hunk 5]
```diff
@@ -1,24 +1,120 @@
...
 export const Summary = () => {
   const { data, isLoading, error } = useSummary();
+  const { quoteAsset } = usePathToMetadata();
 
-  if (error) {
-    return <div className='text-red-600'>Error: {String(error)}</div>;
-  }
+  const change24h = data && {
+    positive: data.current_price >= data.price_24h_ago,
+    change: Number(data.current_price - data.price_24h_ago).toFixed(4),
```
[the remark this answers is not in the document]
Comment (VanishMax) [reply src/pages/trade/ui/summary.tsx]: Thanks, didn't know about this `toFixed` problem. Changed it everywhere to `round`, also in the `shortify` function

**Comment by VanishMax** [comment]
@JasonMHasperhoven do you mind re-reviewing it?

**Review by grod220 — APPROVED** [review]
Let's get this in, can follow up with additional feedback if needed from @JasonMHasperhoven if needed

**Pull request merged**